lib/form/js/repeatable.js

```diff
@@ -0,0 +1,248 @@
+        childNodes.forEach(function(child) {
+          if (child.id != "") {
+            child.id = splitPop(child.id, '--');
+            name = child.getAttribute('name');
+            if(name) {
+              name = splitPop(name, '--');
+              child.setAttribute('name', name);
+            }
+          }
+          else if (child.htmlFor && child.htmlFor != "") {
+            // htmlFor = child.htmlFor.split('--');
+            // htmlFor.pop();
+            // htmlFor.join('--');
+            child.htmlFor = splitPop(child.htmlFor, '--');
+          }
+        });
+        container.removeChild(container.querySelectorAll('#' + input + '-del')[0]);
+      }
+    };
+
+    // Handles add button event
+    var addHandler = function (e) {
+
+      // Holds the first input element of instance
+      var focusChild;
+
+      // selects first container of instance
+      var container = e.currentTarget.parentNode.querySelectorAll('div.form-repeatable')[0];
+
+      // Modify name of initial instance and adds delete button
+      if (allRepeatables[input].length == 1) {
+        var newElement = document.createElement('input');
+        newElement.type = 'button';
+        newElement.value = 'delete';
+        newElement.id = input + '-del';
+        newElement.addEventListener("click", deleteHandler);
+        container.appendChild(newElement);
+        container.id = container.id + '--' + (allRepeatables[input].length-1);
+        var childNodes = container.querySelectorAll('*')
+        childNodes.forEach( function(child) {
+          if (child.id != "") {
+            child.id = child.id + '--' + (allRepeatables[input].length-1);
+            var name = child.getAttribute('name');
+            if (name) {
+              child.setAttribute('name', name +  '--' + (allRepeatables[input].length-1));
+            }
+          }
+          else if (child.htmlFor && child.htmlFor != "") {
+            child.htmlFor = child.htmlFor + '--' + (allRepeatables[input].length-1);
+          }
+        });
+      }
+
+      // deep clones the existing instance
+      var newContainer = container.cloneNode(true);
+      var idMap = {};
+
+      newContainer.id = newContainer.id.split('--')[0] + '--' + allRepeatables[input].length;
+      var childNodes = newContainer.querySelectorAll('*');
+      childNodes.forEach( function (child) {
+        if (child.id != ""){
+          id = splitPop(child.id, '--');
+          var name = child.getAttribute('name');
+          if (name) {
+            name = splitPop(name, '--');
+            child.setAttribute('name', name +  '--' + allRepeatables[input].length);
+          }
+
+          if (idMap.hasOwnProperty(id)){
+            child.id = idMap[id];
+          }
+          else {
+            var newId = id + '--' + allRepeatables[input].length;
+            idMap[id] = newId;
+            child.id = newId;
+          }
+
+          if (child.type == 'text') {
```

Comment:
There could any number of `type`s other than `text` (url, email, etc). This check should be done in a different way.